qemu/tests/mq_change_qnum.py

```diff
@@ -29,30 +29,35 @@ def change_queues_number(session, ifname, q_number, queues_status=None):
         """
         Change queues number
         """
-        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         if not queues_status:
             queues_status = get_queues_status(session, ifname)
 
-        if (q_number != queues_status[1] and q_number <= queues_status[0] and
-                q_number > 0):
-            expect_status = 0
-        else:
-            expect_status = 1
-
+        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         status, output = session.cmd_status_output(mq_set_cmd, safe=True)
         cur_queues_status = get_queues_status(session, ifname)
-        if status != expect_status:
-            err_msg = "Change queues number failed, "
-            err_msg += "current queues set is %s, " % queues_status[1]
-            err_msg += "max allow queues set is %s, " % queues_status[0]
+
+        err_msg = ""
+        expect_q_number = q_number
+        if (q_number != queues_status[1] and q_number <= queues_status[0]
+                and q_number > 0):
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                err_msg = "Param is valid, but change queues failed, "
+        elif cur_queues_status != queues_status:
+            if q_number != queues_status[1]:
+                err_msg = "Param is invalid, "
+            err_msg += "Current queues value is not expected, "
+            expect_q_number = queues_status[1]
+
+        if len(err_msg) > 0:
+            err_msg += "current queues set is %s, " % cur_queues_status[1]
+            err_msg += "max allow queues set is %s, " % cur_queues_status[0]
             err_msg += "when run cmd: '%s', " % mq_set_cmd
-            err_msg += "expect exit status is: %s, " % expect_status
+            err_msg += "expect queues are %s," % expect_q_number
```

Comment:
Use q_number instead of expect_q_number.

Reply:
@xiywang If q_number is a invalid number, it's not the expected queues.

Reply:
@xiywang  For example, current q=4, max_q=4. user input 5. For user's experience, they wish 5. But from testing perspective, the correct expected value should be 4. Does this make sense?

Reply:
I feel like the current logic is not so perfect if there's a product bug, since the purpose of auto test is to find the product bug.
For example: if the max queue is 4, current queue is 4, and the user wants to set it to 5. And let's presume that there's a product bug and the value did changed to 5. With our auto code, we cannot find out this kind of bugs.
Anyway, I don't have a better solution right now. I'll approve this PR for now. And if I come out with other solutions, I'll discuss with you.